sdk/core/Azure.Core.Experimental/tests/MutableJsonDocumentTests.cs

```diff
@@ -117,6 +117,64 @@ public void CanSetPropertyMultipleTimes()
                 jsonString);
         }
 
+        [Test]
+        public void CanSetProperty_JsonWithDotProperty()
+        {
+            string json = """
+                {
+                  "Baz.NotChild" : {
+                     "A" : 3
+                  },
+                  "Foo" : 1,
+                  "Bar" : "Hi!"
+                }
+                """;
+
+            var jd = MutableJsonDocument.Parse(json);
+
+            jd.RootElement.GetProperty("Foo").Set(3);
+
+            Assert.AreEqual(3, jd.RootElement.GetProperty("Foo").GetInt32());
+
+            MutableJsonDocumentWriteToTests.WriteToAndParse(jd, out string jsonString);
+
+            Assert.AreEqual(
+                MutableJsonDocumentWriteToTests.RemoveWhiteSpace("""
+                {
+                  "Baz.NotChild" : {
+                     "A" : 3
+                  },
+                  "Foo" : 3,
+                  "Bar" : "Hi!"
+                }
+                """),
+                jsonString);
+        }
+
+        [Test]
+        public void JsonWithDelimiterIsInvalidJson()
```

Comment:
This is here as protection that the delimiter isn't accidentally changed to something that could break serialization.